student.rb

```diff
@@ -1,12 +1,18 @@
+require_relative 'person'
+
 class Student < Person
   attr_reader :classroom
 
-  def initialize(name, age, parent_permission, classroom)
-    super(name, age, parent_permission)
-    @classroom = classroom
+  def initialize(age, name: 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
   end
 
   def play_hooky
-    '¯\\_(ツ)_/¯'
+    '¯(ツ)/¯'
+  end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.students << self
```

Comment:
- [ ] it would be better to add a guard that checks if a classroom already exists and will prevent repetition like having the same classroom twice that way you will be more reliable
```suggestion
    classroom.students.push(self) unless classroom.students.include?(self)
```

Reply:
Thank you so much.. I have added the guard.